llama-index-core/llama_index/core/readers/file/base.py

```diff
@@ -20,6 +21,32 @@
 from tqdm import tqdm
 
 
+class BaseFilesystemReader(BaseReader, ABC):
+    @abstractmethod
+    def list_files(self, **kwargs) -> List[Path]:
+        """List files in the given filesystem."""
+
+    async def alist_files(self, **kwargs) -> List[Path]:
+        """List files in the given filesystem asynchronously."""
+        return self.list_files(**kwargs)
+
+    @abstractmethod
+    def get_file_info(self, input_file: Path, **kwargs) -> Dict:
+        """Get FS-specific file info that uniquely identifies the file. This call shouldn't imply reading the file."""
+
+    async def aget_file_info(self, input_file: Path, **kwargs) -> Dict:
+        """Get file info that uniquely identifies the file asynchronously. This call shouldn't imply reading the file."""
+        return self.get_file_info(input_file, **kwargs)
+
+    @abstractmethod
+    def read_file(self, input_file: Path, **kwargs) -> List[Document]:
+        """Read file from filesystem and return documents."""
```

Comment:
should this return the file content instead? to separate the file parsing from file loading